pootle/apps/pootle_store/unit/search.py

```diff
@@ -145,22 +149,36 @@ def results(self):
         return self.sort_qs(self.filter_qs(self.units_qs))
 
     def search(self):
-        uid_list = None
-        begin = 0
-        end = 2 * self.chunk_size
-        if self.initial:
-            results = self.results
-            uid_list = list(results.values_list('id', flat=True))
-            if len(self.uids) == 1:
-                if self.uids[0] not in uid_list:
-                    return [], results.none()
-                index = uid_list.index(self.uids[0])
-                begin = max(index - self.chunk_size, 0)
-                end = min(index + self.chunk_size + 1, len(uid_list))
-        elif self.uids:
-            results = self.results.filter(id__in=self.uids)
-        else:
-            # used by export view
-            uid_list = list(self.results.values_list('id', flat=True))
-            results = self.results[:self.chunk_size]
-        return uid_list, results[begin:end]
+        total = self.results.count()
+        start = self.offset
+        if start > total:
+            total, total, total, self.results.none()
+
+        find_unit = (
+            not self.offset
+            and self.language_code
+            and self.project_code
+            and self.filename
+            and self.uids)
+        find_next_slice = (
+            self.previous_uids
+            and self.offset) and True or False
+
+        if find_unit:
+            # find the uid in the Store
+            uid_list = list(self.results.values_list("pk", flat=True))
+            unit_index = uid_list.index(self.uids[0])
+            start = int(unit_index / (2 * self.chunk_size)) * (2 * self.chunk_size)
```

Comment:
we dont return a list of uids any more.

this finds the offset for the client from the uid - in the specific case of translating a store